app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -341,6 +366,16 @@ watchEffect(() => {
     page.value = newPageLimit
   }
 
+  if (rowData.length) {
+    const headers = Object.keys(rowData[0] as object)
+    let obj = {}
+    headers.forEach((header) => {
+      const needsGrouping = rowData.some((row: any) => row[header] > 9999)
+      obj = { ...obj, [header]: needsGrouping }
+    })
```

Comment:
This `obj` update introduces a lot of unnecessary intermediate objects. Doing it by mutable field assignment would avoid that problem. Let's use a proper `Map` instead, which will also force the inserts to be mutable. Vue also properly supports `ref`s for `Map` objects.